examples/computer_vision/cifar10_pytorch_inference/model_def.py

```diff
@@ -0,0 +1,95 @@
+"""
+Perform inference on pretrained CIFAR10 from https://github.com/huyvnphan/PyTorch_CIFAR10
+"""
+
+import tempfile
+from typing import Any, Dict, Sequence, Tuple, Union, cast
+
+import torch
+import torchvision
+from torch import nn
+from torchvision import transforms
+import torchvision.models as models
+
+from determined.pytorch import DataLoader, PyTorchTrial, PyTorchTrialContext
+import resnet
+
+# Constants about the data set.
+IMAGE_SIZE = 32
+NUM_CHANNELS = 3
+NUM_CLASSES = 10
+
+TorchData = Union[Dict[str, torch.Tensor], Sequence[torch.Tensor], torch.Tensor]
+
+
+def accuracy_rate(predictions: torch.Tensor, labels: torch.Tensor) -> float:
+    """Return the accuracy rate based on dense predictions and sparse labels."""
+    assert len(predictions) == len(labels), "Predictions and labels must have the same length."
+    assert len(labels.shape) == 1, "Labels must be a column vector."
+
+    return (  # type: ignore
+        float((predictions.argmax(1) == labels.to(torch.long)).sum()) / predictions.shape[0]
+    )
+
+class CIFARTrial(PyTorchTrial):
+    def __init__(self, context: PyTorchTrialContext) -> None:
+        self.context = context
+        self.download_directory = tempfile.mkdtemp()
+
+        # TODO: Load your trained model
+        self.model = self.context.wrap_model(resnet.resnet18(pretrained=True))
+
+        # IGNORE: Dummy optimizer that needs to be specified but is unused
+        self.optimizer = self.context.wrap_optimizer(torch.optim.RMSprop(
+            self.model.parameters()))
+
+    def train_batch(
+        # IGNORE: No-op train_batch that does not train or generate metrics
+        self, batch: TorchData, epoch_idx: int, batch_idx: int
+    ) -> Dict[str, torch.Tensor]:
+        return {}
+
+    def evaluate_batch(self, batch: TorchData) -> Dict[str, Any]:
+        """
+        Calculate validation metrics for a batch and return them as a dictionary.
+        This method is not necessary if the user defines evaluate_full_dataset().
+        """
+
+        # TODO: Perform your evaluation step
+        batch = cast(Tuple[torch.Tensor, torch.Tensor], batch)
+        data, labels = batch
+        output = self.model(data)
+
+        # TODO: Optionally log or save outputs to persistent store
+        print(output)
+        print(labels)
+        '''
+        with open("/path/to/output.txt", "w+") as f:
```

Comment:
i think it's probably better to write a numpy array instead of writing to a text file.  what we can do is use a custom metric reducer to gather the predictions from across all gpus and then convert that to numpy array and do np.save